### PR TITLE
MB-14757 add and implement factory.BuildDocument

### PR DESCRIPTION
**Pull request by mlm55**
## Summary

This ticket adds a factory to build documents and replaces calls to testdatagen.MakeDocuments functions with the new factory functions.

## Setup to Run Your Code
Run make server_test and make sure tests pass

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/services/moving_expense/moving_expense_deleter_test.go [hunk 1]
```diff
@@ -24,11 +24,7 @@ func (suite *MovingExpenseSuite) TestDeleteMovingExpense() {
 			},
 		})
 
-		expenseDocument := testdatagen.MakeDocument(appCtx.DB(), testdatagen.Assertions{
-			Document: models.Document{
-				ServiceMemberID: serviceMember.ID,
-			},
-		})
+		expenseDocument := factory.BuildDocumentLinkServiceMember(suite.DB(), serviceMember)
```
Comment (mlm55) [inline pkg/services/moving_expense/moving_expense_deleter_test.go:27]: I _think_ `suite.DB` and `appctx.DB` end up being the same. Switched it to suite.DB to match the calls to `BuildServiceMember` and `MakeMinimalPPMShipment` above. I don't feel strongly about it, so let me know if i should stick with `appctx.DB`

Comment (StuTruss) [reply pkg/services/moving_expense/moving_expense_deleter_test.go:27]: yeah I don't have a problem with this as long as tests pass

**Comment by StuTruss** [comment]
everything LGTM, we should probably change the calls for `MakeDefaultDocument` as well in this PR then merge it

File: pkg/services/moving_expense/moving_expense_deleter_test.go (hunk 1 above)
Comment (ahobson) [reply pkg/services/moving_expense/moving_expense_deleter_test.go:27]: I think you are right! We can go one better and remove the `AppContext` parameter from `setupTest` entirely to reduce confusion. I'll do that now

**Pull request merged**